drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-api/src/main/java/org/drools/workbench/screens/scenariosimulation/model/Simulation.java

```diff
@@ -49,11 +61,31 @@ public Scenario getScenarioByIndex(int index) {
     }
 
     public Scenario addScenario() {
+        return addScenario(scenarios.size());
+    }
+
+    public Scenario addScenario(int index) {
+        if (index < 0 || index > scenarios.size()) {
+            throw new IllegalArgumentException(new StringBuilder().append("Index out of range ").append(index).toString());
+        }
         Scenario scenario = new Scenario(simulationDescriptor);
-        scenarios.add(scenario);
+        scenarios.add(index, scenario);
         return scenario;
     }
 
+    public Scenario cloneScenario(int sourceIndex, int targetIndex) {
+        if (sourceIndex < 0 || sourceIndex >= scenarios.size()) {
+            throw new IllegalArgumentException(new StringBuilder().append("SourceIndex out of range ").append(sourceIndex).toString());
+        }
+        if (targetIndex < 0 || targetIndex > scenarios.size()) {
+            throw new IllegalArgumentException(new StringBuilder().append("TargetIndex out of range ").append(targetIndex).toString());
+        }
```

Comment:
Do we really need these index checks in this class? I had a look on [javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#add-int-E-) and it seems to me as not needed.

Reply:
These code comes from https://github.com/kiegroup/drools-wb/pull/927 - that has been approved. Anyway, ask @danielezonca about it.

Reply:
I created these checks to avoid a general IOOB exception so the user/developer will see which index is wrong at least

Reply:
Thanks for answers, will not dig deeper into it. Leaving review for scenario experts.